lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -877,23 +912,20 @@ impl<TGraphLinks: GraphLinks> VectorIndex for HNSWIndex<TGraphLinks> {
                 // - to use HNSW index with filtering condition
 
                 // if exact search is requested, we should not use HNSW index
-                if exact {
-                    let exact_params = params.map(|params| {
-                        let mut params = *params;
-                        params.quantization = Some(QuantizationSearchParams {
-                            ignore: true,
-                            rescore: Some(false),
-                            oversampling: None,
-                        }); // disable quantization for exact search
-                        params
+                if exact || is_hnsw_disabled {
```

Comment:
Do we also want this branch if `vector_storage.available_vector_count() < self.config.full_scan_threshold`, similar to above?

Reply:
This branch is a shortcut to avoid unneeded `estimate_cardinality` call. For unfiltered requests it is not needed at all, but here we can split in 2 cases